examples/all-clusters-app/linux/main.cpp

```diff
@@ -42,6 +47,22 @@ Identify gIdentify1 = {
     EMBER_ZCL_IDENTIFY_IDENTIFY_TYPE_VISIBLE_LED,
 };
 
+Protocols::InteractionModel::Status MatterPreAttributeChangeCallback(const app::ConcreteAttributePath & attributePath, uint8_t mask,
+                                                                     uint8_t type, uint16_t size, uint8_t * value)
+{
+    Protocols::InteractionModel::Status status = Protocols::InteractionModel::Status::Success;
+    if (attributePath.mClusterId == app::Clusters::Thermostat::Id)
+    {
+        status = MatterThermostatClusterServerPreAttributeChangedCallback(attributePath, type, size, value);
```

Comment:
As far as I know the cluster specific callbacks were not yet in place.

Reply:
I see the cluster-specific pre-change callbacks present in the tv-casting app:

```
    const EmberAfGenericClusterFunction chipFuncArrayIasZoneServer[] = {                                                           \
        (EmberAfGenericClusterFunction) emberAfIasZoneClusterServerInitCallback,                                                   \
        (EmberAfGenericClusterFunction) emberAfIasZoneClusterServerMessageSentCallback,                                            \
        (EmberAfGenericClusterFunction) MatterIasZoneClusterServerPreAttributeChangedCallback,                                     \
    };                                                                                                                             \
```

Not sure if it is working though. But if it isn't, we should endeavor to fix it.

Reply:
It should work, assuming the cluster has been added to: https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/templates/app/helper.js#L109

This is something I'm trying to get rid off though but it will takes time, so it sounds fair to reuse the linked mechanism.

Reply:
@bzbarsky-apple removed the hook up through the `MatterPreAttributeChangeCallback`.